### ISSUE TITLE
Simplified user interface

### ISSUE THREAD

**Issue by hcsaustrup** [keepassxreboot/keepassxc#9675]
## Summary

Long time fan, first time caller ;-) I have the dubious honour or being the IT guy in the household, and I'm trying to teach my partner and kids about the importance of security. Naturally I've tried to convince my partner that KeepassXC is key to this, but the user interface was a complete deal breaker for her. Admittedly it is very feature rich and perhaps a little overwhelming to some. Now my kids are coming to a point where they need passwords for things, and their dad is getting a little annoyed with having to type in long auto-generated passwords from his cellphone every time they need to identify. Again, the KeepassXC experience - as it is - would probably be a little discouraging for them.

Basically what I'm looking for, is a simplified mostly read-only interface with all the add/create/clone/update and advanced features hidden. Any thoughts on this?

**Comment by droidmonkey** [comment]
We are very unlikely to provide an interface like this without redoing the entire application. For better or worse we inherit the "flow" of the application from KeePass original and KeePassX (our direct predecessor). 

What you can do to limit the amount of "noise" visible is to hide the toolbar (view menu), hide the group panel (drag it all the way to the left), and create a bunch of entries with your family to show them how to do that. Basically, all their entries would go into the root group.

**Comment by michaelk83** [comment]
They may be more comfortable with one of the mobile apps that are compatible with KeePass.
(Though of course, those can't automate password entry on the PC.)

**Comment by MarkWieczorek** [comment]
As much as I love KeePassXC, I have to agree with the original post. I tried to get my non-tech savy spouse to switch from 1password, but the user interface is just too "feature rich".

The recommendations of @droidmonkey are a great start, and they would be very simple to implement : All you would need to do is create a toggle in the settings for "advanced/simple user interface". When KeepassXC is run for the first time, the user would decide if they wanted the simplified or advanced UI (which could be changed later). The simple user interface would

* hide the vast majority of the application settings: all settings would be on a single page.
* limit the entry fields that are shown to title, username, password, URL and TOTP
* by default, hide the "enter additional credentials (if necessary)" dialog from the lock screen.
* hide the groups and tags panels by default
* perhaps remove some of the toolbar settings, like "perform autotype", and open database, 
* clicking on password generator would automatically copy the proposed password to the clipboard

Adding such a simplified UI would probably increase the number of KeePassXC users by more than a factor or 2.

**Comment by droidmonkey** [comment]
That also increases our overhead by a factor of 2. It's basically a whole other program to maintain. Whenever we adjust the UI or add a component we'll have to consider the simple ui. That is why I answered this is likely never going to happen I'm a realistic way. It is already very hard for us to maintain a consistent and relatively sane UX.

**Comment by michaelk83** [comment]
I think the situation can be gradually improved without necessarily creating two interfaces:
- Simplify the entry edit widget:
  - Implement #8228, which would allow users to quickly select a preset of fields for a new entry.
  - Move custom attributes to a (+) button on the main edit page. This then allows the default template to be simplified, since ~~more advanced~~ less common fields would be accessible through the (+) button.
  - Maybe replace the side bar with a selection of the entry layouts, allowing the layout to be switched easily. Any non-empty fields would be be added to the selected layout same as they would through the (+) button. A `...` option at the bottom of the layout selection bar would then show the current side bar, where you can add attachments, set auto-type, etc. (Btw, attachments could be moved to the (+) button as well.)
- Simplify password workflows. There are a number of open issues on that.
- Add a "Show Sidebar" entry to the View menu.
- On the Unlock Database widget, hide the additional credentials behind a toggle.

And so on. Each of these can be implemented separately as its own task. For settings, there could be a simplified settings page with just the most common settings, but I think that settings are an advanced feature in general.

I think it's a good idea to always keep ease of use in mind. In web design, there is a concept called "mobile first". Where, you start by defining the minimal feature set for low-end mobile devices, and then add more features on more capable devices. I think a similar principle can be applied here, but thinking of user capabilities rather than device capabilities. Start from the minimal useful UI in each area, but allow more advanced features to be easily accessed when needed.

**Comment by droidmonkey** [comment]
This is covered by #4244
